core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -621,14 +621,17 @@ object SparkSubmit {
     if (isKubernetesCluster) {
       childMainClass = "org.apache.spark.deploy.kubernetes.submit.Client"
       if (args.isPython) {
-        childArgs += args.primaryResource
-        childArgs += "org.apache.spark.deploy.PythonRunner"
-        childArgs += args.pyFiles
+        childArgs ++= Array("--primary-py-file", args.primaryResource)
```

Comment:
Wondering if it makes sense for `(childArgs, childClasspath, sysProps, childMainClass)` to be modeled as a case class with builder pattern, for similar reasons.  Tangentially, mutable collections can be hazardous if not handled carefully - a case-class pattern using immutable collections might be worthwhile, given the complexity of the environment constructions